src/Uno.UWP/UI/Input/GestureRecognizer.Manipulation.cs

```diff
@@ -0,0 +1,283 @@
+						if (forceUpdate || IsSignificant(cumulative))
+						{
+							_state = ManipulationStates.Started;
+							_lastPublished = _currents;
+
+							_recognizer.ManipulationStarted?.Invoke(
+								_recognizer,
+								new ManipulationStartedEventArgs(_deviceType, _currents.Center, cumulative));
+						}
+
+						break;
+
+					case ManipulationStates.Started when _recognizer.ManipulationUpdated == null:
+						_lastPublished = _currents;
+						break;
+
+					case ManipulationStates.Started:
+						// Even if Scale and Angle are expected to be default when we add a pointer (i.e. forceUpdate == true),
+						// the 'delta' and 'cumulative' might still contains some TranslateX|Y compared to the previous Pointer1 location.
+						var delta = GetDelta(_lastPublished);
+						if (forceUpdate || IsSignificant(delta))
+						{
+							_lastPublished = _currents;
+
+							_recognizer.ManipulationUpdated.Invoke(
+								_recognizer,
+								new ManipulationUpdatedEventArgs(_deviceType, _currents.Center, delta, GetDelta(), isInertial: false));
+						}
+						break;
+				}
+			}
+
+			private ManipulationDelta GetDelta() => GetDelta(_origins);
+			private ManipulationDelta GetDelta(Points from)
+			{
+				var isTranslateXEnabled = (_settings & (GestureSettings.ManipulationTranslateX | GestureSettings.ManipulationTranslateRailsX)) != 0;
+				var isTranslateYEnabled = (_settings & (GestureSettings.ManipulationTranslateY | GestureSettings.ManipulationTranslateRailsY)) != 0;
+				var translateX = isTranslateXEnabled ? _currents.Center.X - from.Center.X : 0;
+				var translateY = isTranslateYEnabled ? _currents.Center.Y - from.Center.Y : 0;
+
+				double rotate;
+				float scale, expansion;
+				if (!_currents.HasPointer2)
+				{
+					rotate = 0;
+					scale = 1;
+					expansion = 0;
+				}
+				else
+				{
+					var isRotateEnabled = (_settings & (GestureSettings.ManipulationRotate | GestureSettings.ManipulationRotateInertia)) != 0;
+					var isScaleEnabled = (_settings & (GestureSettings.ManipulationScale | GestureSettings.ManipulationScaleInertia)) != 0;
+
+					rotate = isRotateEnabled ? _currents.Angle - from.Angle : 0;
+					scale = isScaleEnabled ? _currents.Distance / from.Distance : 1;
+					expansion = isScaleEnabled ? _currents.Distance - from.Distance : 0;
+				}
+
+				return new ManipulationDelta
+				{
+					Translation = new Point(translateX, translateY),
+					Rotation = (float)MathEx.ToDegreeNormalized(rotate),
+					Scale = scale,
+					Expansion = expansion
+				};
+			}
+
+			private bool IsSignificant(ManipulationDelta delta)
+				=> Math.Abs(delta.Translation.X) >= MinManipulationDeltaTranslateX
+				|| Math.Abs(delta.Translation.Y) >= MinManipulationDeltaTranslateY
+				|| delta.Rotation >= MinManipulationDeltaRotate // We used the ToDegreeNormalized, no need to check for negative angles
+				|| Math.Abs(delta.Expansion) >= MinManipulationDeltaExpansion;
+
+			// WARNING: This struct is ** MUTABLE **
+			private struct Points
```

Comment:
Why choose a struct here ?

Reply:
At the beginning it was only a way to aggregate a bunch of private field, then it evolves to something similar to a class. Is there any perf reason to change it to a class?